fluidsynth/src/synth/fluid_chan.h

```diff
@@ -41,7 +41,7 @@ struct _fluid_channel_t
   int sfont_bank_prog;                  /**< SoundFont ID (bit 21-31), bank (bit 7-20), program (bit 0-6) */
   fluid_preset_t* preset;               /**< Selected preset */
 
-  int key_pressure;                     /**< MIDI key pressure */
+  int key_pressure[128];                /**< MIDI polyphonic key pressure */
```

Comment:
You are right, the pressure value is always in the range 0-127, so char makes absolute sense.